File: ios/Status/Activity.swift
//
//  Activity.swift
//  Runner
//
//  Created by ender on 29/7/23.
//

import Foundation

struct Activity: Decodable, Hashable {
    let id: String
    let partner: String?
    let birthControl: String?
    let partnerBirthControl: String?
    let date: Date
    let location: String?
    let notes: String?
    let duration: Int
    let orgasms: Int
    let partnerOrgasms: Int
    let place: String?
    let initiator: String?
    let rating: Int
    let type: String?
    let mood: String?
    let practices: [IdName]
}

struct IdName: Hashable, Codable {
    let id: String
    let name: String
}

File: android/app/src/main/kotlin/works/end/LoveLust/glance/Activity.kt
package works.end.LoveLust.glance

import kotlinx.serialization.Serializable

@Serializable
data class Activity(
    val id: String? = null,
    val partner: String? = null,
    val birth_control: String? = null,
    val partner_birth_control: String? = null,
    val date: String,
    val location: String? = null,
    val notes: String? = null,
    val duration: Int,
    val orgasms: Int,
    val partner_orgasms: Int,
    val place: String? = null,
    val initiator: String? = null,
    val rating: Int,
    val type: String = "SEXUAL_INTERCOURSE",
    val practices: List<String>? = null,
    val mood: String? = null,
    val ejaculation: String? = null,
    val watched_porn: Boolean,
)